src/java/org/apache/cassandra/io/sstable/metadata/MetadataCollector.java

```diff
@@ -100,6 +92,7 @@ public static StatsMetadata defaultStatsMetadata()
                                  null,
                                  false,
                                  true,
+                                 MutationIdRanges.NONE,
```

Comment:
Yikes so this is used by `MetadataSerializer` which silently allows you to fail to read the stats metadata file if doesn't exist. The default values here are generally conservative to ensure things behave correctly like setting it to not repired, min and max values for timestamps and keys.

Mutation ID ranges should probably similarly conservative in some way that ensures that we don't mark this as "repaired" until it must have been repaired by definition.

Reply:
Thought about this some more - I think we should not tolerate missing StatsMetadata files anymore and log a warning. This tolerance was added ~12 years ago for the 2.1 beta when StatsMetadata was introduced, but every supported upgrade path now should have StatsMetadata already present.

For CASSANDRA-20381, we could refuse to mark SSTables as reconciled if they are for tracked keyspaces and have MutationIdRanges.NONE, since presumably these SSTables existed before a migration to mutation tracking, and the migration path should probably do an IR and mark SSTables as repaired.